include/ocpp/v201/component_state_manager.hpp

```diff
@@ -23,12 +23,102 @@ struct FullConnectorStatus {
 
     /// \brief Translates the individual state to an Available/Unavailable/Occupied/Reserved/Faulted state
     /// This does NOT take into account the state of the EVSE or CS,
-    /// and is intended to be used internally by the ComponentStateManager.
+    /// and is intended to be used internally by the ComponentStateManagerInterface.
     ConnectorStatusEnum to_connector_status();
 };
 
+class ComponentStateManagerInterface {
+public:
+    virtual ~ComponentStateManagerInterface();
+
+    virtual void set_cs_effective_availability_changed_callback(
```

Comment:
Please add the method docs for this function

Reply:
This change is just from refactoring the ComponentStateManager into an interface so that we could mock it. So, while we'd love to have documentation, we don't really have a good grasp on what this is supposed to do so that we can document it.

Reply:
Turns out we missed the original docs. Copied those over.